### PR TITLE
feat: [CDS-84451]: add proxy config

### PR DESCRIPTION
**Pull request by mteodor**
## Comment Triggers

<details>
  <summary>PR Check triggers</summary>
  
- Build: `trigger build`
- Sub Category Field Check: `trigger subcategoryfieldcheck`
- gitleaks: `trigger gitleaks`

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -83,6 +83,6 @@ require (
 	gotest.tools/v3 v3.3.0 // indirect
 )
 
-// replace github.com/harness/harness-go-sdk => ../harness-go-sdk
+replace github.com/harness/harness-go-sdk => ../harness-go-sdk
```
Comment (manavjot-harness) [inline go.mod:86]: comment this back.

File: internal/service/platform/gitops/repository_certificates/data_source_gitops_repo_cert_test.go [hunk 2]
```diff
@@ -64,3 +70,38 @@ func testAccDataSourceRepoCert(id string, accountId string, name string, agentId
...
+				upsert = true
+				certificates {
+					metadata {
+
+					}
+					items {
+						server_name = "bitbucket.org"
+						cert_type = "ssh"
+						cert_sub_type = "ssh-rsa"
+						cert_data = "QUFBQUIzTnphQzF5YzJFQUFBQUJJd0FBQVFFQXViaU44MWVEY2FmcmdNZUx6YUZQc3cya052RWNxVEtsL1ZxTGF0L01hQjMzcFp5MHkzckpadG5xd1IycU9PdmJ3S1pZS2lFTzFPNlZxTkVCeEt2SkplbENxMGRUWFdUNXBiTzJnRFhDNmg2UURYQ2FIbzZwT0hHUFV5K1lCYUdRUkd1U3VzTUVBU1lpV3VuWU4wdkNBSThRYVhuV01YTk1kRlAzakhBSkgwZURzb2lHbkxQQmxCcDRUTm02cllJNzRuTXpnejNCOUlpa1c0V1ZLK2RjOEtaSlpXWWpBdU9SVTNqYzFjL05Qc2tEMkFTaW5mOHYzeG5mWGV1a1Uwc0o1TjZtNUU4VkxqT2JQRU8rbU4ydC9GWlRNWkxpRnFQV2MvQUxTcW5Nbm5od3JOaTJyYmZnL3JkL0lwTDhMZTNwU0JuZTgrc2VlRlZCb0dxekhNOXlYdz09"
```
Comment (mankrit-singh) [inline internal/service/platform/gitops/repository_certificates/data_source_gitops_repo_cert_test.go:95]: are these secrets we should avoid committing?

File: internal/service/platform/gitops/agent_yaml/data_source_gitops_agent_deploy_yaml_test.go [hunk 3]
```diff
@@ -66,6 +66,13 @@ func testAccDataSourceGitopsAgentDeployYaml(agentId string, accountId string, ag
...
...EJBUUVGQUFPQ0FROEFNSUlCQ2dLQ0FRRUF2UXJRR3JpdXN5OG5Hbk1hWjhvUk9nd1NiN05OdDNkM3llb1oKV0JmV2ZNU0xhWXpwdjcvL0Noc2lSdzlFUTNKcFN1SlF0bUx4SDdsZHcwNVZyY0M4VTBQOWFEWlZ1Q1ljOStSTwpiRUF6MmtwVUFkcUw4N29uYkN6OVkweERwTmJIZDJOaGtkZGF6ME9DbDJJOU10MGdSTk9ZT0N1RXliZS90TStvCmR3WVdMTnYrMXJGb2NKbEFJYjZ0Z2t1MldoZUNNYXlsV1Jqc1U1VldkNXdUTitUWW9GN25YWVhjWmY3cHhtekYKUytTY3l3NDN4M2hsU0E3RzNsZnY2Ri9VOWY5YVpZU0ZVUktMenByQllXaWpUR3F6Mm94M3VDUWF0MFlpTkdxMwpoS2RZY2N4UVJJQnl3dDEyR3RmaUFja0l6NnpKVVdoZzJDN1cxZkIzN2ZBNVBWOFRsUUlEQVFBQm8xTXdVVEFkCkJnTlZIUTRFRmdRVXY2ZUx3QytURjBHSzlyamx0TVEwMzh3NFQzb3dId1lEVlIwakJCZ3dGb0FVdjZlTHdDK1QKRjBHSzlyamx0TVEwMzh3NFQzb3dEd1lEVlIwVEFRSC9CQVV3QXdFQi96QU5CZ2txaGtpRzl3MEJBUXNGQUFPQwpBUUVBSGVyWGc1a2hEVkxpWG9ZSmpRMnhTQ2xoQlVIdGdSTGJ6R25ZekJ1R3VseTh1UW9BZ1dLZU1kM0pjSk93CmJ4K3c3NzFsUzFNbmdENEhiK0ZXWWxkdE5xUHZQa2c3RXZKb2lFMHQzSElzck02WXdyNDUvNHBSZVBMWU1paSsKV0FqTFhOZGVuUUUwVlFvY2pzKzN4M0QyK0FOYitRUTFxVTAzYVhiSEVRQzdmU2k1Y2pPMjd1aWRocVoyNEtHbQpJOE0vN0VmRWhWR01LeStKYnd3WGdYaVZvQ2FHK3QyUTRHS3NETlNJRlNWVEpsT1JPbXBYUUVZdmxIMmRndzdkClVieWRZR1l0TXlBY3hSSHVTWCtwMVNXSFZabDJ1TjMvd1AzWWt1M1kwbXFkdzVPdk1Lc1htbkJFNGFtb3BvQm0KWFV4V2F6U3YxaG5iaGdMWkRWVHk4VmRYY0E9PQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg=="
```
Comment (mankrit-singh) [inline internal/service/platform/gitops/agent_yaml/data_source_gitops_agent_deploy_yaml_test.go:69]: same here

File: internal/service/platform/gitops/repository_certificates/data_source_gitops_repo_cert_test.go (hunk 2 above)
Comment (mteodor) [reply internal/service/platform/gitops/repository_certificates/data_source_gitops_repo_cert_test.go:95]: these are the values for known hosts that are copied from the argocd config map, these are public values

File: internal/service/platform/gitops/agent_yaml/data_source_gitops_agent_deploy_yaml_test.go (hunk 3 above)
Comment (mteodor) [reply internal/service/platform/gitops/agent_yaml/data_source_gitops_agent_deploy_yaml_test.go:69]: this is dummy self signed certificate

**Review by mankrit-singh — APPROVED** [review]

**Pull request merged**